hal/src/rtl872x/spi_hal.cpp

```diff
@@ -322,6 +322,13 @@ class Spi {
             u32 rtlClockDivider = 256;
             clockDivToRtlClockDiv(config_.clockDiv, &rtlClockDivider);
             SSI_SetBaudDiv(SPI_DEV_TABLE[rtlSpiIndex_].SPIx, rtlClockDivider);
+
+            // Set sample delay for SPI0@50MHz
+            if (rtlClockDivider == 2 && SPI_DEV_TABLE[rtlSpiIndex_].SPIx == SPI0_DEV) {
+                SSI_SetSampleDelay(SPI_DEV_TABLE[rtlSpiIndex_].SPIx, 0x1);
+            } else {
+                SSI_SetSampleDelay(SPI_DEV_TABLE[rtlSpiIndex_].SPIx, 0x0);
```

Comment:
Is it working fine with other `rtlClockDivider`?

Reply:
This issue only happens when using the high speed SPI, in case of affecting low speed SPI, we only set it for SPI0@50MHz (`rtlClockDivider ==2`).

Reply:
How do you know the sample delay being `0` is working with other low speed SPI? This is also the new introduced change to the low speed SPI.

Reply:
Because `0x0` is the default value
<img width="787" alt="Screenshot 2023-08-01 at 15 18 39" src="https://github.com/particle-iot/device-os/assets/7424522/a71a668e-263c-4278-945b-c7bbeb49f04d">